### PR TITLE
[FEATURE] Update diagnostic checklist to do linting checks

### PR DESCRIPTION
**Pull request by kenwade4**
Changes proposed in this pull request:
- Update diagnostic checklist to do linting checks
- Make linting check be a part of experimental checks instead of production checks
- Also move camel_to_snake func to great_expectations/util.py

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py
Comment (cdkini) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py]: Nitpick - could we just import `os` to be consistent with the rest of the codebase?

File: great_expectations/expectations/expectation.py
Comment (cdkini) [inline great_expectations/expectations/expectation.py:1539]: Should `check_linting` be a public method? I feel little iffy invoking a private method if we can avoid it.

File: great_expectations/util.py
Comment (cdkini) [inline great_expectations/util.py:99]: Could we give these more descriptive names? What do `p1` and `p2` do here?

File: great_expectations/util.py
Comment (cdkini) [inline great_expectations/util.py:130]: Could we type annotate the args and return here? Also, could we add a few unit tests? I trust your regex skills a lot more than I do mine but I think a few tests to gain coverage would be great!

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py
Comment (cdkini) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py]: Could we add type annotations to all variable declarations throughout?
```suggestion
        sub_messages: List[dict] = []
        passed: bool = False
        black_ok: bool = False
        isort_ok: bool = False
        file_and_class_names_ok: bool = False
```

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py
Comment (cdkini) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py]: Will this ever fail? What if group doesn't have 1?

Nitpick - could we split up this logic into separate lines? Perhaps the match on one and the group on the other?

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py
Comment (cdkini) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:488]: love it!

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py
Comment (cdkini) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py]: If this regex is only used in the single method, could we perhaps do this within the function? I'd rather clean up the global namespace if at all possible, even if it means being slightly less performant.

File: great_expectations/util.py
Comment (cdkini) [reply great_expectations/util.py:130]: Oops now seeing you just ported this over - will defer to you here.

**Review by cdkini — APPROVED** [review]
LGTM - left comments around type annotations and general style that I'd love resolved but I don't want to block on that.

File: great_expectations/expectations/expectation.py
Comment (kenwade4) [reply great_expectations/expectations/expectation.py:1539]: All the staticmethod/classmethod decorated methods in the ExpectationDiagnostics class start with `_check_`

File: great_expectations/util.py
Comment (kenwade4) [reply great_expectations/util.py:99]: Just moved as-is from great_expectations/expectations/expectation.py

File: great_expectations/util.py
Comment (kenwade4) [reply great_expectations/util.py:130]: Just moved as-is from great_expectations/expectations/expectation.py

**Pull request merged**
